subset/password/brute_test/src/main/java/TelnetSocket.java

```diff
@@ -0,0 +1,317 @@
+    telnetClient = null;
+    telnetClient = new TelnetClient();
+    System.out.println("Starting Telnet Connection");
+    try {
+      telnetClient.connect(host, connectionPort);
+      System.out.println("Connected");
+    } catch (Exception e) {
+      System.err.println(e);
+    }
+  }
+
+  private String normalizeLineEnding(byte[] bytes, char endChar) {
+
+    List<Byte> bytesBuffer = new ArrayList<Byte>();
+
+    int countBreak = 0;
+    int countESC = 0;
+
+    for (int i = 0; i < bytes.length; i++) {
+      if (bytes[i] != 0) {
+        switch (bytes[i]) {
+          case 8:
+            // backspace \x08
+            break;
+          case 10:
+            // newLineFeed \x0A
+            countBreak++;
+            bytesBuffer.add((byte) endChar);
+            break;
+          case 13:
+            // carriageReturn \x0D
+            countBreak++;
+            bytesBuffer.add((byte) endChar);
+            break;
+          case 27:
+            // escape \x1B
+            countESC = 2;
+            break;
+          case 33:
+            // character:!
+            break;
+          default:
+            if (countESC == 0) {
+              if (countBreak > 1) {
+                int size = bytesBuffer.size();
+                for (int x = 0; x < countBreak - 1; x++) {
+                  bytesBuffer.remove(size - 1 - x);
+                }
+                countBreak = 0;
+              }
+              bytesBuffer.add(bytes[i]);
+            } else {
+              countESC--;
+            }
+            break;
+        }
+      }
+    }
+
+    String bytesString = "";
+
+    for (Byte byteBuffer : bytesBuffer) {
+      bytesString = bytesString + (char) (byte) byteBuffer;
+    }
+
+    return bytesString;
+  }
+
+  public void readData() {
+    int bytesRead = 0;
+
+    inputStream = telnetClient.getInputStream();
+
+    while (telnetClient.isConnected()) {
+      inputStream = telnetClient.getInputStream();
+
+      try {
+        byte[] buffer = new byte[1024];
+
+        bytesRead = inputStream.read(buffer);
+        if (bytesRead > 0) {
+          String rawData = normalizeLineEnding(buffer, '\n');
+          rxQueue.add(rawData);
+        } else {
+          try {
+            Thread.sleep(100);
+          } catch (InterruptedException e) {
+            System.err.println("InterruptedException readData:" + e.getMessage());
+          }
+        }
+      } catch (IOException e) {
+        System.err.println("Exception while reading socket:" + e.getMessage());
+      }
+    }
+  }
+
+  public void gatherData() {
+    StringBuilder rxData = new StringBuilder();
+    String rxGathered = "";
+    int rxQueueCount = 0;
+    int rxTempCount = 0;
+
+    while (telnetClient.isConnected()) {
+      try {
+        if (rxQueue.isEmpty()) {
+          Thread.sleep(100);
+          rxQueueCount++;
+          if (debug) {
+            System.out.println("rxQueue.isEmpty:" + rxQueueCount);
+          }
+          if (rxQueueCount > 70) {

```

Comment:
what is the logic behind this?  Can you describe in a simple English sentence what's this means?    "If there's a bunch of data then throw it all out" doesn't seem right.